From 90e65f30abef5841ade5ab1494e8c5bf79f2baf4 Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Sun, 12 Feb 2012 16:44:21 +0100
Subject: [PATCH 12/12] a11y: Mark cells as defunct when deleting them

Just unreffing isn't enough, atk-bridge might have a reference to the
object still.
---
 gtk/a11y/gtktreeviewaccessible.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/gtk/a11y/gtktreeviewaccessible.c b/gtk/a11y/gtktreeviewaccessible.c
index 2807e41..8bec291 100644
--- a/gtk/a11y/gtktreeviewaccessible.c
+++ b/gtk/a11y/gtktreeviewaccessible.c
@@ -97,6 +97,7 @@ gtk_tree_view_accessible_get_data_quark (void)
 static void
 cell_info_free (GtkTreeViewAccessibleCellInfo *cell_info)
 {
+  gtk_accessible_set_widget (GTK_ACCESSIBLE (cell_info->cell), NULL);
   g_object_unref (cell_info->cell);
 
   g_free (cell_info);
-- 
1.7.9

